distribution/lib/Standard/Base/0.0.0-dev/src/Data/Ordering.enso

```diff
@@ -23,42 +23,38 @@ from project.Data.Boolean import all
     Define conversion function `Comparable.from` for your `type` and return pointer to
-    another `type` that satisfies either of the following two definitions:
+    another `type` that satisfies the following definition:
 
     ```
-    type Ordered_Comparator T
-        is_ordered = True
+    type Comparator T
         compare : T -> T -> (Ordering|Nothing)
         hash : T -> Integer
-
-    type Unordered_Comparator T
-        is_ordered = False
-        equals : T -> T -> (Boolean|Nothing)
-        hash : T -> Integer
     ```
 
-    Or `Incomparable` in case that the type `T` should not be compared at all.
-
     Every type must provide exactly one comparator, i.e., any method of form
     `Comparable.from (_:My_Type)` must always return the same comparator type.
-    Note that if there is an implicit `Default_Unordered_Comparator` assigned
-    for every type by default.
+    Note that there is an implicit `Default_Comparator` assigned for every
+    type by default.
+
+    A comparator has to implement `compare` and `hash` methods. `compare x y` method
+    returns `Ordering.Less` if x is ordered before y, `Ordering.Equal` if x is equal
+    to y, `Ordering.Greater` if x is ordered after y, and `Nothing` if x and y are
+    _incomparable_ or not equal.
 
-    Note that there has to be `is_ordered` method defined which returns a Boolean
-    indicating that the comparator is ordered. This is currently needed as there is
-    no way to define interfaces in Enso.
+    Note that _incomparable_ and _not equal_ is not, strictly speaking, the same
+    relation. But for the sake of simplicity of the `compare` method, they are the
+    same relations.
 
-    An _unordered comparator_ has to implement both `equals` and `hash` to define
-    a _partial_ custom equality. By _partial_, we mean that every instance of the type
-    has to be either equal, not equal, or undefined, which is represented by the type signature
-    of `equals` - when a `Boolean` is returned, the two instances are either equal
-    or not equal, when `Nothing` is returned, the instances are incomparable.
+    The signature of the `compare` method is designed so that it enables some comparator
+    to provide a _partial ordering_ for a specific type.
 
-    An _ordered comparator_ has `compare` method instead of `equals` method, that is
-    expected to return `Ordering` signaling that an instance of the type is either
-    less than, equal to, or greater than the other instance, or `Nothing` signaling
-    that the two instances are incomparable. This relation may also be _partial_, meaning
-    that some of the instances may be incomparable.
+    _unordered_ comparator is a comparator, whose `compare` method only returns `Ordering.Equal`,
+    or `Nothing`. _ordered_ comparator is a comparator that returns only `Ordering.Equal`,
+    `Ordering.Less`, or `Ordering.Greater`. _partially ordered_ comparator is a comparator
+    that can return anything.
```

Comment:
Let's start sentences with Capital letters.

I don't like the distinction. There can be partially ordered comparators where for some values it returns Less/Greater and for others it is Nothing. There can be ones where it always is Less/Greater or always is Nothing, but that does not exhaust the possible options. I think the paragraphs above are enough of an explanation.

Reply:
You are right. I originally wrote this paragraph so that _unordered_ and _ordered_ comparators terminology is properly defined. But I guess that you are right that it only complicates the understanding. Removing.